client/lib/purchases/index.js

```diff
@@ -27,6 +30,7 @@ function getPurchasesBySite( purchases ) {
 				domain: currentValue.domain,
 				id: currentValue.siteId,
 				name: currentValue.siteName,
+				slug: sites.getSite( currentValue.siteId ).slug,
```

Comment:
It's only called in a single place under `client/me/purchases/list/index.jsx`, and *I think* that page is only loaded after we have the sites. But apart from that, there is no guarantee. I have come to think that it might be a better idea to move the function to somewhere else, because a dependency to `sites-list` breaks many tests since `sites-list` depends on DOM and we need to add `react-test-env-setup` to many tests to get them running.